### ISSUE TITLE
Overlay using CDN

### ISSUE THREAD

**Issue by 3d0** [palantir/blueprint#1255]
### Bug report

- __Package version(s)__: Core from 1.11 to 1.20, using react-with-addons from 15.3.1 to 15.6.1
- __Browser and OS versions__: Chrome 59

#### Steps to reproduce

1. Build a simple page with a buildless react (with addons)
1. Add the CDN support of Blueprint
1. add an overlay to the view

#### Actual behavior

Overlay doesn't appear at all. The console yields error: 
Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in. Check the render method of `Blueprint.Overlay`.

#### Expected behavior

An overlay appearing?

#### Evidence

https://jsfiddle.net/yjsb0np9/3/

**Comment by adidahiya** [comment]
You have a JS error in your console; make sure to include all of Blueprint's dependencies:

```
Uncaught TypeError: Cannot read property 'className' of undefined
    at unpkg.com/@blueprintjs/core@1.20.0:2383
    at mapSingleChildIntoContext (npmcdn.com/react@15.6.1/dist/react-with-addons.js:1021)
    at traverseAllChildrenImpl (npmcdn.com/react@15.6.1/dist/react-with-addons.js:3759)
    at traverseAllChildren (npmcdn.com/react@15.6.1/dist/react-with-addons.js:3854)
    at mapIntoWithKeyPrefixInternal (npmcdn.com/react@15.6.1/dist/react-with-addons.js:1041)
    at Object.mapChildren [as map] (npmcdn.com/react@15.6.1/dist/react-with-addons.js:1063)
    at Overlay.render (unpkg.com/@blueprintjs/core@1.20.0:2381)
    at npmcdn.com/react-dom@15.6.1/dist/react-dom.js:5257
    at measureLifeCyclePerf (npmcdn.com/react-dom@15.6.1/dist/react-dom.js:4537)
    at ReactCompositeComponentWrapper._renderValidatedComponentWithoutOwnerOrContext (npmcdn.com/react-dom@15.6.1/dist/react-dom.js:5256)
```

**Comment by 3d0** [comment]
I know that, but which one?
if you look my example I added all the libraries (react-with-addons, react-dom, classnames, tether, blueprintjs) and all the css (normalize and blueprint) according to the instructions given [here](http://blueprintjs.com/docs/#blueprint.cdn-consumption)

**Comment by adidahiya** [comment]
Ah, ok, I see the problem, sorry about that. There are two things going on:

1. `Overlay` expects a JSX element as its child, not a simple string (easy to fix in your JSFiddle, low priority for us to fix in Blueprint since all real use cases for Overlay will use an element as its only child)
2. `React.addons.CSSTransitionGroup` is not being picked up correctly as a webpack external in the Blueprint core JS bundle, so when Overlay tries to return a `CSSTransitionGroup` in its `render` method, it ends up returning `undefined` and React gets unhappy. Not sure if this is a webpack bug but it's definitely a bug in CDN consumption. This reference would work if it was `root["React"].addons.CSSTransitionGroup`:
  ![image](https://user-images.githubusercontent.com/723999/27344530-91e590c8-55b4-11e7-977c-cc89162a0136.png)

**Comment by 3d0** [comment]
Let me add to this that because of this error you cannot use DateRangeInput (or, in fact, any object that depends on an Overlay)

**Comment by giladgray** [comment]
i'm confident this has been resolved with the latest 2.0 CDN bundles. many folks have gotten it working on codesandbox: #1526
